examples/evaluation/huggingface-evaluation.ipynb

```diff
@@ -0,0 +1,1340 @@
+      "        \n",
+      "\n",
+      "Example Input:\n",
+      "How is MLflow related to Databricks?\n",
+      "\n",
+      "Example Output:\n",
+      "MLflow is a product created by Databricks to enhance the efficiency of machine learning processes.\n",
+      "\n",
+      "Additional information used by the model:\n",
+      "key: targets\n",
+      "value:\n",
+      "MLflow is an open-source platform for managing the end-to-end machine learning (ML) lifecycle. It was developed by Databricks, a company that specializes in big data and machine learning solutions. MLflow is designed to address the challenges that data scientists and machine learning engineers face when developing, training, and deploying machine learning models.\n",
+      "\n",
+      "Example score: 4\n",
+      "Example justification: The output provided by the model is mostly correct. It correctly identifies that MLflow is a product created by Databricks. However, it does not mention that MLflow is an open-source platform for managing the end-to-end machine learning lifecycle, which is a significant part of its function. Therefore, while the output is mostly accurate, it has a minor omission, which is why it gets a score of 4 according to the grading rubric.\n",
+      "        \n",
+      "\n",
+      "You must return the following fields in your response one below the other:\n",
+      "score: Your numerical score for the model's answer_correctness based on the rubric\n",
+      "justification: Your step-by-step reasoning about the model's answer_correctness score\n",
+      "    )\n"
+     ]
+    }
+   ],
+   "source": [
+    "from mlflow.metrics.genai import answer_correctness\n",
+    "\n",
+    "answer_correctness_metric = answer_correctness(model=\"openai:/gpt-4\")\n",
+    "\n",
+    "print(answer_correctness_metric)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "ac9cdf24-d024-4543-a292-371bb301041f",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## Evaluate"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We need to set our OpenAI API key, since we are using GPT-4 for our LLM-judged metrics."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "3b56fb6b-2272-4ddf-b297-b222a13efb77",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"redacted\""
```

Comment:
We should not recommend setting a private key in a notebook at any time. Can we convert this to an instruction on how to set the key safely?

Something along the lines of:

In order to set your private key safely, please be sure to either export your key through a command-line terminal for your current instance, or, for a permanent addition to all user-based sessions, configure your favored environment management configuration file (i.e., `.bashrc`, `.zshrc`) to have the following entry:

``` bash
OPENAI_API_KEY=<your openai API key>
```